Dockerfile

```diff
@@ -1,33 +1,48 @@
-FROM phusion/baseimage:0.9.18
+FROM phusion/baseimage:0.9.19
 ENV DEBIAN_FRONTEND noninteractive
 
-RUN echo "debconf shared/accepted-oracle-license-v1-1 select true" | debconf-set-selections
-RUN echo "debconf shared/accepted-oracle-license-v1-1 seen true" | debconf-set-selections
-
 # First, install add-apt-repository and bzip2
-RUN apt-get update
-RUN apt-get -y install software-properties-common python-software-properties bzip2 unzip openssh-client git lib32stdc++6 lib32z1
-
 # Add oracle-jdk8 to repositories
-RUN add-apt-repository ppa:webupd8team/java
-
-# Update apt
-RUN apt-get update
-
 # Install oracle-jdk8
-RUN apt-get -y install oracle-java8-installer
+# Install Ruby
+RUN echo "debconf shared/accepted-oracle-license-v1-1 select true" | debconf-set-selections && \
```

Comment:
Docker creates an image after every command and therefore doing the install and cleanup in one command results in less for every build to download.